components/soyosource_virtual_meter/soyosource_virtual_meter.cpp

```diff
@@ -112,11 +112,11 @@ int16_t SoyosourceVirtualMeter::calculate_power_demand_negative_measurements_(in
     return this->max_power_demand_;
   }
 
-  if (power_demand > this->min_power_demand_) {
-    return power_demand;
+  if (power_demand < this->min_power_demand_) {
+    return this->min_power_demand_;
```

Comment:
Could you move this fix into a separate PR?

Reply:
ok: https://github.com/kev300/esphome-soyosource-gtn-virtual-meter/tree/min_power_demand_fix